src/components/SearchToolbar/CategoryButton.js

```diff
@@ -44,24 +45,40 @@ function CategoryButton(props: Props) {
   );
 
   return (
-    <Button
-      onClick={() => props.onClick(category)}
-      className={className}
-      onFocus={props.onFocus}
-      aria-label={showCloseButton ? t`Remove ${props.name} Filter` : props.name}
-    >
-      <IconButton isHorizontal={showCloseButton} caption={props.name} hasCircle={props.hasCircle}>
-        {icon}
-      </IconButton>
-      {showCloseButton && <CloseIcon />}
-    </Button>
+    <RouteConsumer>
+      {context => {
+        const params = { ...context.params };
```

Comment:
0: Maybe do this, as the Category gets replaced anyway

```suggestion
        const { prevCategory, ...params } = context.params;
```

Reply:
Personally I like the descriptive delete more than extracting an object property without using it. I'm always worried somebody working with the code thinks its a mistake and removes the property from the object spreading. What do you think?

Reply:
Fine by me :)